darts/models/forecasting_model.py

```diff
@@ -62,6 +62,13 @@ def fit(self, series: TimeSeries) -> None:
         self.training_series = series
         self._fit_called = True
 
+        if hasattr(self, "trend"):
+            raise_if(
+                self.trend and self.trend != "c" and self.training_series.has_dummy_index,
```

Comment:
I think here instead of check for `self.trend` we should probably call a function `self.supports_dummy_index()` or something like that. This way the super-class doesn't have to know about the specifics of the implementations (e.g. checking `self.trend != 'c'`. It is rather in `AutoARIMA` that we should do this check. We can have a default `ForecastingModel.supports_dummy_index()` always returning `True`, which could be overridden in `AutoARIMA`.